clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -353,7 +354,12 @@ public void verifyHeartbeatSent() throws Exception {
         // initial fetch
         client.prepareResponseFrom(fetchResponse(tp0, 0, 0), node);
 
-        consumer.poll(0);
+        // We need two update calls:
+        // 1. the first call "sends" the metadata update requests
+        // 2. the second one gets the response we already queued up
+        consumer.internalUpdateAssignmentMetadataIfNeeded(Duration.ZERO);
+        consumer.internalUpdateAssignmentMetadataIfNeeded(Duration.ZERO);
+
```

Comment:
I hope that comment makes sense. Alternatively, we could use one call with a non-zero timeout (such as 1ms), but I liked the double async call, since it seems to be a tighter bound on the behavior.